File: Core/UserQueryOptions.cs
namespace Core;
public class UserQueryOptions : BaseQueryOptions
{
    public string? PartialUserName { get; set; }
    public IEnumerable<int>? FilterRoleIds { get; set; }
    public bool? IsBlocked { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
}

File: client/src/app/shared/models/QueryOptions/UserQueryOptions.ts
import AvailableUserRole from '../enums/AvailableUserRole';
import BaseQueryOptions from './BaseQueryOptions';

export default interface UserQueryOptions extends BaseQueryOptions {
  partialUserName: string | undefined;
  filterRoles: AvailableUserRole[] | undefined;
  isBlocked: boolean | undefined;
  startDate: Date | undefined;
  endDate: Date | undefined;
}
